frontend/javascripts/oxalis/model/reducers/annotation_reducer.ts

```diff
@@ -178,17 +180,27 @@ function AnnotationReducer(state: OxalisState, action: Action): OxalisState {
           max: V3.toArray(V3.round(V3.add(action.center, halfBoxExtent))),
         };
       }
-      let newBoundingBox: UserBoundingBox;
+      let newUserBoundingBox: UserBoundingBox;
       if (action.newBoundingBox != null) {
-        newBoundingBox = {
+        newUserBoundingBox = {
           ...newBoundingBoxTemplate,
           ...action.newBoundingBox,
         };
       } else {
-        newBoundingBox = newBoundingBoxTemplate;
+        newUserBoundingBox = newBoundingBoxTemplate;
       }
 
-      const updatedUserBoundingBoxes = [...userBoundingBoxes, newBoundingBox];
+      // Ensure the new bounding box is within the dataset bounding box.
+      const datasetBoundingBox = getDatasetBoundingBox(state.dataset);
+      const newBoundingBox = new BoundingBox(newUserBoundingBox.boundingBox);
+      const newBoundingBoxWithinDataset = newBoundingBox.intersectedWith(datasetBoundingBox);
+      // Only update the bounding box if the bounding box overlaps with the dataset bounds.
+      // Else the bounding box is completely outside the dataset bounds -> in that case just keep the bounding box and let the user cook.
+      if (newBoundingBoxWithinDataset.getVolume() > 0) {
+        newUserBoundingBox.boundingBox = newBoundingBoxWithinDataset.toBoundingBoxType();
+      }
```

Comment:
hm, now that I tested the branch, I can create the bbox outside of the dataset by clicking and dragging. from the code, I wouldn't have expected that :thinking: can you clarify this?

Reply:
That is because when creating a bbox via drag, a initial bbox of size [1,1,1] is created at the mouse pointer and afterwards the bbox is resized with update actions. These update actions via drag are not interfered with.
Here are the functions creating the initial [1,1,1] bbox:
https://github.com/scalableminds/webknossos/blob/b866aef631137f23f9d1eebe90bee261d0a0b3af/frontend/javascripts/oxalis/controller/combinations/tool_controls.ts#L572-L586
And
https://github.com/scalableminds/webknossos/blob/b866aef631137f23f9d1eebe90bee261d0a0b3af/frontend/javascripts/oxalis/controller/combinations/bounding_box_handlers.ts#L231-L264

This is, why it is working :)

Reply:
thanks for the explanation!